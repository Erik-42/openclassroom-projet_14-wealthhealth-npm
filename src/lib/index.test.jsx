import React from "react";
import { cleanup, render, screen } from "@testing-library/react";
import { afterEach, describe, expect, test } from "vitest";
import ModaleReactErik42 from "./index";
import ModaleErik42 from "./components/modaleErik42/modaleErik42";

describe("ModaleReact", () => {
	afterEach(() => {
		cleanup();
	});
	// Test du rendu sans props
	test("Le composant ModaleReact se rend correctement sans props", () => {
		render(<ModaleErik42 showModale={true} />);
		const modaleElement = screen.getByTestId("modal");
		expect(modaleElement).toBeTruthy();
	});
	// Test de la prop greetee
	test("La prop greetee fonctionne correctement", () => {
		render(<ModaleReactErik42 greetee="Erik-42" />);
		const greeteeText = screen.getByText(/Erik-42/i);
		expect(greeteeText).toBeTruthy();
	});
	// Test avec les paramètres de la modale
	test("ModaleReact fonctionne avec les paramètres de la modale", async () => {
		render(
			<ModaleErik42
				id="testModale"
				showModale={true}
				closeModale={() => {}}
				parameter={{ backgroundColor: "red" }}
				message="Message de test"
			/>
		);

		const modaleElement = screen.getByTestId("modal");
		expect(modaleElement).toBeTruthy();
		const messageElement = screen.getByText(/Message de test/i);
		expect(messageElement).toBeTruthy();
	});
	// Test supplémentaire pour les interactions ou comportements
	test("Ouverture de la modal et affichage du message", () => {
		render(
			<ModaleErik42
				id="testModale"
				showModale={true}
				closeModale={() => {}}
				parameter={{ backgroundColor: "red" }}
				message="Message de test"
			/>
		);
		const modaleElement = screen.getByTestId("modal");
		expect(modaleElement).toBeTruthy();
		const messageElement = screen.getAllByText(/Message de test/i);
		expect(messageElement).toBeTruthy();
	});
});
