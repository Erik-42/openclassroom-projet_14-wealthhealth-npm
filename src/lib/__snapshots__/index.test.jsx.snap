// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelloWorld > HelloWorld component renders correctly 1`] = `
<div>
  Hello, 
  World
  !
</div>
`;

exports[`HelloWorld > The greetee prop works 1`] = `
<div>
  Hello, 
  Universe
  !
</div>
`;

exports[`ModaleReact > La prop greetee fonctionne correctement 1`] = `
<div>
  <div>
    Modale React, 
    Erik-42
  </div>
</div>
`;

exports[`ModaleReact > Le composant ModaleReact se rend correctement sans props 1`] = `
<div>
  <div>
    Modale React, 
  </div>
</div>
`;

exports[`ModaleReact > ModaleReact fonctionne avec les param√®tres de la modale 1`] = `
<div>
  <div>
    Modale React, 
  </div>
</div>
`;
